identity-verification/module_2/2.3 Authentication IDV API Example.ipynb

```diff
@@ -0,0 +1,466 @@
+    "    'Name':selfie_image}},\n",
+    "                                   Attributes=['ALL'])\n",
+    "response"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 4. Authentication 1:n Identification with SearchFaceByImage\n",
+    "-----\n",
+    "\n",
+    "For a given input image, first detects the largest face in the image, and then searches the specified collection for matching faces. The operation compares the features of the input face with faces in the specified collection.\n",
+    "\n",
+    "<div class=\"alert alert-info\"><strong> Note </strong>\n",
+    "    \n",
+    "\n",
+    "- SimilarityThreshold - The minimum level of confidence in the face matches that a match must meet to be included in the FaceMatches array. \n",
+    "\n",
+    "- QualityFilter - A filter that specifies a quality bar for how much filtering is done to identify faces. Filtered faces aren't compared. If you specify AUTO, Amazon Rekognition chooses the quality bar. If you specify LOW, MEDIUM, or HIGH, filtering removes all faces that do not meet the chosen quality bar. \n",
+    "\n",
+    "</div>\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "response = rek_client.search_faces_by_image(\n",
+    "    CollectionId=collection_name,\n",
+    "    Image={\n",
+    "        'S3Object': {\n",
+    "            'Bucket': bucket_name,\n",
+    "            'Name': selfie_image,\n",
+    "        }\n",
+    "    },\n",
+    "    MaxFaces=1,\n",
+    "    FaceMatchThreshold=90,\n",
+    "    QualityFilter='AUTO'\n",
+    ")\n",
+    "response\n",
+    "\n",
+    "external_image_id = response[\"FaceMatches\"][0][\"Face\"][\"ExternalImageId\"]\n",
+    "similarity_score =  response[\"FaceMatches\"][0][\"Similarity\"]\n",
+    "face_id = response[\"FaceMatches\"][0][\"Face\"][\"FaceId\"]\n",
+    "\n",
+    "print(\"-- FaceID          : {} \".format(face_id))\n",
+    "print(\"-- Similarity      : {} \".format(similarity_score))\n",
+    "print(\"-- ExternalImageID : {} \\n\".format(external_image_id))\n",
+    "\n",
+    "print(json.dumps(response, indent=3))  \n",
+    "\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1:n images in a collection?\n",
+    "\n",
+    "what happens when we have more than one image of a user in a collection? lets check it out! \n",
+    "\n",
+    "<div class=\"alert alert-info\"><strong> Steps </strong>\n",
+    "    \n",
+    "\n",
+    "- Here we are going to index several images into our collection \n",
+    "- then search the collection with a new selfie \n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "images_to_index = [\"face_52_0.jpeg\", \"face_52_1.jpeg\", \"face_52_2.jpeg\"]\n",
+    "new_selfie = \"face_52_3.jpeg\"\n",
```

Comment:
This isn't used before being redeclared on line 363. Perhaps remove one of them?